src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,128 @@
+
+import duke.command.ByeCommand;
+import duke.command.Command;
+import duke.command.DeadlineCommand;
+import duke.command.DeleteCommand;
+import duke.command.DoneCommand;
+import duke.command.ErrorCommand;
+import duke.command.EventCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.command.TodoCommand;
+
+/**
+ * Parser object that parses all the input from user to commands understood by Duke
+ */
+public class Parser {
+    private static DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd/MM/uuuu HHmm");
+    private static String[] parsed;
+
+    private static int parseTaskNo(String parse) {
+        int taskNo = Integer.parseInt(parse) - 1;
+        return taskNo;
+    }
+
+    private static String parseTodoTask() throws IllegalArgumentException {
+        if (parsed.length < 2 || parsed[1].equals("")) {
+            throw new IllegalArgumentException();
+        }
+        return parsed[1];
+    }
+
+    private static String[] parseEventTask(String command) throws
+            IllegalArgumentException,
+            ArrayIndexOutOfBoundsException,
+            StringIndexOutOfBoundsException {
+        String[] details = new String[2];
+        String desc = command.split(" /at ")[0].substring(6);
+        if (desc.equals("")) {
+            throw new IllegalArgumentException();
+        }
+        details[0] = desc;
+        details[1] = command.split(" /at ")[1];
+        if (details[1].equals("")) {
+            throw new ArrayIndexOutOfBoundsException();
+        }
+        return details;
+    }
+
+    private static String[] parseDeadlineTask(String command) throws
+            IllegalArgumentException,
+            ArrayIndexOutOfBoundsException,
+            StringIndexOutOfBoundsException {
+        String[] details = new String[2];
+        String desc = command.split(" /by ")[0].substring(9);
+        if (desc.equals("")) {
+            throw new IllegalArgumentException();
+        }
+        details[0] = desc;
+        details[1] = command.split(" /by ")[1];
+        if (details[1].equals("")) {
+            throw new ArrayIndexOutOfBoundsException();
+        }
+        return details;
+    }
+
+    private static LocalDateTime parseDate(String dateTime) throws DateTimeException {
+        return LocalDateTime.parse(dateTime, formatter);
+    }
+
+    private static String parseKeyword(String input) throws IllegalArgumentException {
+        String keyword = input.substring(4).strip().toLowerCase();
+        if (keyword.equals("")) {
+            throw new IllegalArgumentException();
+        }
+        return keyword;
+    }
+
+    /**
+     * Parses input line by line
+     *
+     * @param input the input of the user
+     * @return returns a command type to be executed in duke.run()
+     */
+    public static Command parseCommands(String input) {
+        parsed = input.split(" ");
+        try {
+            switch (parsed[0]) {
+            case ByeCommand.COMMAND:
```

Comment:
Perhaps you could use a set of enums with the Command names instead? Like LIST, DONE, DELETE. This way there would be no need to include/access the COMMAND constant in each command.